doc/check.texi

```diff
@@ -1314,8 +1316,70 @@ two environment variables available that offers this ability,
 the name of the suite and/or test case you want to run. These
 environment variables can also be a good integration tool for
 running specific tests from within another tool, e.g. an IDE.
-  
-@node Testing Signal Handling and Exit Values, Looping Tests, Selective Running of Tests, Advanced Features
+
+@node Selecting Tests Based on Arbitrary Tags, Testing Signal Handling and Exit Values, Selective Running of Tests, Advanced Features
+@section Selecting Tests Based on Arbitrary Tags
+
+@vindex CK_INCLUDE_TAGS
+@vindex CK_EXCLUDE_TAGS
+
+It can also be useful to be able to dynamically select or exclude
+groups of tests to be run based on criteria other than the suite or
+testcase name. For example certain test cases can be tagged as
+requiring a long run time and so quick sanity tests can be run that
+exclude all test cases with such a tag. Alternately tags may be used
+where tests cover multiple functional areas in order to indicate which
+areas a test case covers. Tests can then be run that include all test
+cases for a given set of areas.
+
+A tag is a string of characters excluding spaces.
+
+Tags are applied to a testcase by passing a space separated list of
+tags to @code{tcase_set_tags} once it has been created. For example in
+function @code{make_tagged_suite()} in @file{check_check_tags.c} - 
+
+@example
+@verbatim
+    Suite *s;
+
+    TCase *red, *blue, *purple, *yellow, *black;
+
+    s = suite_create("Check Tag Filtering");
+
+    red = tcase_create("Red");
+    tcase_set_tags(red, "Red");
+    suite_add_tcase (s, red);
+    tcase_add_test(red, red_test1);
+
+    blue = tcase_create("Blue");
+    tcase_set_tags(blue, "Blue");
+    suite_add_tcase (s, blue);
+    tcase_add_test(blue, blue_test1);
+
+    purple = tcase_create("Purple");
+    tcase_set_tags(purple, "Red Blue");
+    suite_add_tcase (s, purple);
+    tcase_add_test(purple, purple_test1);
+
+@end verbatim
+@end example
+
+Once test cases are tagged then there are two environment variables
+available for selecting testcases based on these tags
+@code{CK_INCLUDE_TAGS} and @code{CK_EXCLUDE_TAGS}. These can be set to
+a space separated list of tag names. If @code{CK_INCLUDE_TAGS} is set
+then only testcases which include at least one tag in common with
+@code{CK_INCLUDE_TAGS} will be run. If @code{CK_EXCLUDE_TAGS} is set
+then testcases with one tag in common with @code{CK_EXCLUDE_TAGS} will
+not be run. In cases where both @code{CK_INCLUDE_TAGS} and
+@code{CK_EXCLUDE_TAGS} match then the test will be excluded.
+
+Both @code{CK_INCLUDE_TAGS} and @code{CK_EXCLUDE_TAGS} can be
```

Comment:
You mean one that explictly uses the env variables ? There is already an include_exclude suite that tests the correct filtering - its just that it does not use the env vars - it would certainly be easy enough to clone it to produce an env var version as well if you think that's useful. Probably should just for completeness.